third_party/move/move-compiler-v2/src/flow_insensitive_checkers.rs

```diff
@@ -0,0 +1,203 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Flow-insensitive checks can be done on the AST.
+//!
+//! Warnings about Unused parameter and local variable
+//!   "Unused assignment or binding for local 's'. Consider removing, replacing with '_' or prefixing with '_' (e.g., '_r_ref')
+
+use codespan_reporting::diagnostic::Severity;
+use move_model::{
+    ast::{ExpData, TempIndex},
+    model::{GlobalEnv, Loc, NodeId, Parameter},
+    symbol::Symbol,
+};
+use std::{collections::BTreeSet, iter::Iterator};
+
+/// Warns about all parameters and local variables that are unused.
+pub fn check_for_unused_vars_and_params(env: &mut GlobalEnv) {
+    for module in env.get_modules() {
+        for func in module.get_functions() {
```

Comment:
You should only do this for target modules, otherwise you report warnings/errors in dependencies.

Reply:
Oops, good find.  Not sure how I lost that.